src/services/audit-events-service.ts

```diff
@@ -0,0 +1,197 @@
+import { addMetric } from '../commons/metrics';
+import { transitionConfiguration } from './account-states/config';
+
+const appConfig = AppConfigService.getInstance();
+
+const sqsClient = tracer.captureAWSv3Client(
+  new SQSClient({
+    region: appConfig.awsRegion,
+    maxAttempts: 2,
+    requestHandler: new NodeHttpHandler({ requestTimeout: 5000 }),
+  }),
+);
+
+export class AuditEvents {
+  constructor(
+    private egressEventTransitionType: TxMAEgressEventTransitionType,
+    private ingressEventName: EventsEnum,
+    private ingressTxMAEvent: TxMAIngressEvent,
+    private accountStateEngineOutput?: AccountStateEngineOutput,
+  ) {}
+
+  /**
+   * Function that sends Audit Events to the TxMA Egress queue.
+   * @returns - Response from sending the message to the Queue.
+   */
+  public async send(): Promise<SendMessageCommandOutput | undefined> {
+    logger.debug('sendAuditEvent function.');
+
+    if (this.eventShouldBeIgnored()) return;
+
+    const timestamp = getCurrentTimestamp();
+
+    const txmaEvent = <TxMAEgressEvent>{
+      timestamp: timestamp.seconds,
+      event_timestamp_ms: timestamp.milliseconds,
+      event_timestamp_ms_formatted: timestamp.isoString,
+      component_id: COMPONENT_ID,
+      event_name: this.extractEgressEventName(),
+      user: { user_id: this.ingressTxMAEvent.user.user_id },
+      extensions: this.buildExtensions(),
+    };
+
+    const input = { MessageBody: JSON.stringify(txmaEvent), QueueUrl: appConfig.txmaEgressQueueUrl };
+
+    try {
+      logger.debug('Attempting to send TxMA event to the queue.');
+      const response = await sqsClient.send(new SendMessageCommand(input));
+      addMetric(MetricNames.PUBLISHED_EVENT_TO_TXMA);
+      return response;
+    } catch (error) {
+      addMetric(MetricNames.ERROR_PUBLISHING_EVENT_TO_TXMA);
+      logger.error('An error happened while trying to send the audit event to the TxMA queue.', { error: error });
+    }
+  }
+
+  /**
+   * Function to check if an event should be sent to TxMA
+   */
+  private eventShouldBeIgnored() {
+    if (!this.accountStateEngineOutput) return false;
+
+    const accountState = this.accountStateEngineOutput.stateResult;
+
+    return (
+      userLedActionList.includes(this.ingressEventName) &&
+      !accountState.suspended &&
+      !accountState.blocked &&
+      !accountState.reproveIdentity &&
+      !accountState.resetPassword &&
+      this.egressEventTransitionType === TxMAEgressEventTransitionType.TRANSITION_IGNORED
+    );
+  }
+
+  private extractEgressEventName() {
+    const prefix =
+      this.ingressEventName === EventsEnum.OPERATIONAL_FORCED_USER_IDENTITY_REVERIFICATION
+        ? 'AIS_NON_FRAUD_'
+        : 'AIS_EVENT_';
+
+    return (prefix + this.egressEventTransitionType) as TxMAEgressEventName;
+  }
+
+  /**
+   * Helper function to build extension object based on the type of event
+   * @returns - TxMAEgressExtensions object
+   */
+  private buildExtensions() {
```

Comment:
just out of curiosity, is there a reason for storing all of this info as class variables? :) I personally maybe find it a bit hard to read with all the `this` references, and I am also wondering if there may be some small efficiency gains in having this class as a singleton and just have those class variables as parameters to a method :)